src/test/java/com/answerdigital/benhession/academy/answerkingweek2/controllers/CategoryControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package com.answerdigital.benhession.academy.answerkingweek2.controllers;
+
+import com.answerdigital.benhession.academy.answerkingweek2.request.AddCategoryRequest;
+import com.answerdigital.benhession.academy.answerkingweek2.request.UpdateCategoryRequest;
+import com.answerdigital.benhession.academy.answerkingweek2.services.CategoryService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@ContextConfiguration(classes = {CategoryController.class})
+@ExtendWith(SpringExtension.class)
+class CategoryControllerTest {
+    @Autowired
+    private CategoryController categoryController;
+
+    @MockBean
+    private CategoryService categoryService;
+
+    /**
+     * Method under test: {@link CategoryController#addCategory(AddCategoryRequest)}
+     */
+    @Test
+    void testAddCategory() throws Exception {
```

Comment:
I think we can simplify it to this form: 
    @Test
    @WithMockUser("paul")
    void testAddCategory() throws Exception {
        ObjectMapper objectMapper = new ObjectMapper();
        
        mockMvc.perform(get("/")
               .content(objectMapper.writeValueAsString(new AddCategoryRequest("Name", "The characteristics of someone or something")))
               .contentType(MediaType.APPLICATION_JSON))
               .andExpect(MockMvcResultMatchers.status().isNotFound());
    }

what do you think?